### PR TITLE
Get messages

### PR DESCRIPTION
**Pull request by rottabonus**
# Description

Based on top of #76 which introduced some UI elements to show messages. Now added fetching messages and sending a new message.

### New features:

- Message fetching logic similar than in the app-project
   1. First fetches contacts
   2. Fetch latest 10 messages for each contact (in chunks of 40 contacts at a time, because of queryString limitations)
   3. Poll new messages after every n-seconds
   4. Can fetch more recent messages by scrolling to top of the conversation
   
- Removed eslint-plugin that enforces object-keys to be in alphabetic order
    - It just made the reducer-objects to be more unreadable
    -  That rule/plugin did not provide much value IMO
   
 ### Trello [Trello-card](https://trello.com/c/getsZ8fD/769-chat-page-showing-message-history)
  
  
### Videodemo:

https://github.com/sos-lapsikyla/ylitse-web/assets/34128180/5d17b47a-9617-480d-82ad-8448cd2fb169

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

No tests implemented

- [ ] Unittest
- [ ] Cypress e2e -tests

# Caveats?

Its just a start, for example the components need to be split more. A lot of features are missing, like setting unread-messages to a "read"-state

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by rottabonus** [comment]
Added one commit after making a new self review for this PR.. =P
Saw some minor things I wanted to fix that I did not see two weeks ago. 
Mostly naming, but also:

- Fetch the contacts and messages when user logs in whatever route they are! => It means you cannot simulate the same stuff you see on the demovideo, because the messages are being loaded instantly, but its better =P

**Review by kuosmark — COMMENTED** [review]
Good job! Nice last-minute refactorings too!

**Comment by kuosmark** [comment]
> Fetch the contacts and messages when user logs in (before they go to chat-page) => It means you cannot simulate the same stuff you see on the demovideo, because the messages are being loaded instantly, but this is how we need to do it

Very nice! Are all chats loaded on login? Or is there a limit if there is for example 200 chats? Does this mean that the user never sees the loading spinners in the left-side chat menu?

**Comment by rottabonus** [comment]
> Very nice! Are all chats loaded on login? Or is there a limit if there is for example 200 chats? Does this mean that the user never sees the loading spinners in the left-side chat menu?

Messages are fetched as described in the **PR in bulletpoints.** 

So if user has 400 different conversations, then the initial messages would be divided to 10 separate requests. The requests load pretty fast, but user might be able to quickly go to the chat-tab. 
Or if you are authenticated already and then just refresh the page, then you would see spinning.

File: src/features/Chat/components/Window/index.tsx [hunk 1]
```diff
@@ -1,32 +1,79 @@
...
+  const archiveChat = () => {
+    // this is fine
+  };
+
+  const blockChat = () => {
+    // this is fine
+  };
+
+  const restoreChat = () => {
+    // this is fine
```
[the remark this answers is not in the document]
Comment (rottabonus) [reply src/features/Chat/components/Window/index.tsx:51]: Not sure, I just removed the dummy logic there was before and instead put these placeholders to wait

File: src/hooks/useScrollToTop.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { RefObject, useCallback, useEffect, useState } from 'react';
```
[the remark this answers is not in the document]
Comment (rottabonus) [reply src/hooks/useScrollToTop.ts:1]: Yeah I made some changes to the logic of the hook => renamed => and then I guess forgot the original file there

File: src/features/Chat/components/Window/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kuosmark) [reply src/features/Chat/components/Window/index.tsx:51]: Okay, I added them to the backlog! We can estimate them later when we go through the tasks.

[the remark this answers is not in the document]
Comment (rottabonus) [reply src/features/Chat/components/Window/index.tsx:51]: I also moved sending and receiving message to the review, because they are part of this actually

**Review by kuosmark — APPROVED** [review]

**Pull request merged**